addons/binding/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/A5_11/A5_11_04.java

```diff
@@ -189,7 +193,39 @@ protected State convertToStateImpl(String channelId, String channelTypeId, State
             case CHANNEL_INSTANTPOWER:
                 return getPowerMeasurementData();
             case CHANNEL_TOTALUSAGE:
-                return getEnergyMeasurementData();
+                State value = getEnergyMeasurementData();
+
+                EnOceanChannelTotalusageConfig c = config.as(EnOceanChannelTotalusageConfig.class);
+
+                if (c.validate && (value instanceof QuantityType) && (currentState instanceof QuantityType)) {
```

Comment:
This part is the same for all three files. Can you factor this out into its own utility class? That should drop this PRs code line changes by a lot.

Reply:
Thanks, I have refactored this into a separate class.